src/coreclr/jit/jiteh.cpp

```diff
@@ -640,6 +640,19 @@ bool Compiler::bbIsHandlerBeg(const BasicBlock* block)
     return (ehDsc != nullptr) && ((block == ehDsc->ebdHndBeg) || (ehDsc->HasFilter() && (block == ehDsc->ebdFilter)));
 }
 
+// bbIsFuncletBeg() returns true if "block" is the start of a handler or filter region,
+// and if the handler/filter is a funclet.
+//
+bool Compiler::bbIsFuncletBeg(const BasicBlock* block)
+{
+    if (UsesFunclets() && fgFuncletsCreated)
```

Comment:
Should we even be asking this if `fgFuncletsCreated` is false?

Reply:
Perhaps rearrange so we `assert(fgFuncletsCreated);` under `if(UsesFunclets())`?

Reply:
When I started working on this, we had some paths where we would check for funclet start blocks before and after funclets were created, like in `fgCanCompactBlock`. I've since been able to remove these checks, so we can make the `fgFuncletsCreated` check an invariant.